int32 heat
int32 CO2
